compiler/rustc_const_eval/src/transform/check_consts/qualifs.rs

```diff
@@ -74,6 +74,10 @@ pub trait Qualif {
         adt: AdtDef<'tcx>,
         args: GenericArgsRef<'tcx>,
     ) -> bool;
+
+    /// Returns `true` if this `Qualif` behaves sructurally for pointers and references:
+    /// the pointer/reference qualifies if and only if the pointee qualifies.
+    fn deref_structural<'tcx>(cx: &ConstCx<'_, 'tcx>) -> bool;
```

Comment:
all impls of this are `false` 😆

Reply:
Yes, true. But I didn't want to hard-code that, who knows which qualifs we get in the future.
Also, I think naming it explicitly this way makes the code a lot more clear.

Reply:
ok, please document this before someone optimizes it